packages/components/src/components/FileUploader/FileUploader.tsx

```diff
@@ -0,0 +1,259 @@
+import { SystemProp, Theme, useDown } from "@localyze-pluto/theme";
+import isEmpty from "lodash/isEmpty";
+import React, { ReactElement } from "react";
+import * as HeroOutlineIcons from "@heroicons/react/24/outline";
+import { Box } from "../../primitives/Box";
+import { Text } from "../../primitives/Text";
+import { Anchor } from "../Anchor";
+import { Button, ButtonProps } from "../Button";
+import { Icon } from "../Icon";
+import { HelpText } from "../HelpText";
+import { ProgressBar } from "../ProgressBar";
+import { FileUploaderButtonProps } from "./FileUploaderButton";
+
+export interface FileUploaderProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  /** The title of the file */
+  label: string;
+
+  /** The max size of a file */
+  maxFileSize?: string;
+
+  /** The file name */
+  fileName?: string;
+
+  /** The file url when the file is uploaded */
+  fileUrl?: string;
+
+  /** The size of the file */
+  fileSize?: string;
+
+  /** An error message */
+  errorMessage?: string;
+
+  /** The upload progress */
+  progress?: number;
+
+  /** The call to action button */
+  children: ReactElement<ButtonProps | FileUploaderButtonProps>;
```

Comment:
I know that I'm always asking this because I thought we wanted mandatory fields before the optional ones. But I don't know what the advantage of having it as well is 😅 What do you think?
My only point is that we can have it organized.